### PR TITLE
Infrastructure for PR publishing

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Adds the ability to publish PRs to GitHub Packages for testing purposes.  
The publishing is controlled by a checkbox in a bot comment, which is ticked by default for maintainers, and can be ticked (or unticked) by any maintainer.  
In this repository, a MDK will also be generated with a visible installer link in the comment (more information on that can be found [here](https://github.com/neoforged/action-pr-publishing?tab=readme-ov-file#mdk-generation)).

An example of this system in action can be found [here](https://github.com/ForgeForce/NeoForge/pull/3#issuecomment-1869779480). I would appreciate some testing of the MDK setup script by people running non-Windows systems.

Questions and more information can be found [in the GU PR](https://github.com/neoforged/GradleUtils/pull/17) which this PR depends on.

**The dependency on the GU PR is virtual and not functional (it serves the sole purpose of making sure that there won't be differences between the actual workflow in this repository and the template).** This PR works and can be reviewed. 

For maintainers: leave the merging up to me.

### CODE REVIEW

File: .github/workflows/publish-prs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request_target:
+    types:
+      - opened
+
+permissions:
+  packages: write
+
+jobs:
+  publish-prs:
+    if: true
```
Comment (shartte) [inline .github/workflows/publish-prs.yml:23]: Hmmmmmm?

Comment (Matyrobbrt) [reply .github/workflows/publish-prs.yml:23]: See https://github.com/neoforged/GradleUtils/blob/feature%2Fpr-publishing/src%2FactionsTemplate%2Fresources%2F.github%2Fworkflows%2Fpublish-prs.yml#L20

Comment (shartte) [reply .github/workflows/publish-prs.yml:23]: Heh, these files should have a comment pointing to where hey were generated from...

**Review by sciwhiz12 — DISMISSED** [review]

**Review by embeddedt — DISMISSED** [review]
Very nice; it will be extremely useful for testing API reworks before they are officially shipped.

Also, the MDK setup script works on Linux.

**Review by XFactHD — DISMISSED** [review]
The script provided by the action should mention that the `-L` option needs to be removed from the `curl` invocation for it to work in PowerShell on Windows, but other than that :shipit:

**Review by TheCurle — APPROVED** [review]

**Review by embeddedt — APPROVED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Review by XFactHD — APPROVED** [review]

**Pull request merged**